mailserver/mailserver.go

```diff
@@ -243,9 +266,9 @@ func (s *WMailServer) processRequest(peer *whisper.Peer, lower, upper uint32, bl
 
 	for i.Next() {
 		var envelope whisper.Envelope
-		err = rlp.DecodeBytes(i.Value(), &envelope)
-		if err != nil {
-			log.Error(fmt.Sprintf("RLP decoding failed: %s", err))
+		decodeErr := rlp.DecodeBytes(i.Value(), &envelope)
+		if decodeErr != nil {
```

Comment:
I think in case `DecodeBytes` returns an error, this method will return empty values for ret and error.
Shouldn't we return an error  when an error decoding the envelope is produced?

Reply:
@adriacidre we just log the error but continue with the loop because if there are other envelopes they need to be sent anyway.